test/unit/backend_test.rb

```diff
@@ -0,0 +1,98 @@
+require 'helper'
+
+describe 'Backend' do # rubocop:disable Metrics/BlockLength
+  let(:backend) { Inspec::Backend.create(Inspec::Config.mock) }
+
+  describe 'create' do # rubocop:disable Metrics/BlockLength
+    it 'accepts an Inspec::Config' do
+      backend.is_a?(Inspec::Backend::Base).must_equal true
+    end
+
+    it 'raises an error if no transport backend can be found' do
+      Train.stub :create, nil do
+        err = proc { backend }.must_raise RuntimeError
+        err.message.must_equal "Can't find transport backend 'mock'."
+      end
+    end
+
+    it 'raises an error if no connection can be created' do
+      mock_transport = Minitest::Mock.new
+      mock_bad_connection = Minitest::Mock.new
+      mock_transport.expect :nil?, false
+      mock_transport.expect :connection, mock_bad_connection
+      mock_bad_connection.expect :nil?, true
+
+      Train.stub :create, mock_transport do
+        err = proc { backend }.must_raise RuntimeError
+        err.message.must_equal "Can't connect to transport backend 'mock'."
+      end
+    end
+
+    it 'enables/disables caching based on `config[:backend_cache]`' do
+      cache_enabled_config = Inspec::Config.new(backend_cache: true)
+      cache_enabled_backend = Inspec::Backend.create(cache_enabled_config)
+      cache_enabled_backend.backend.cache_enabled?(:file).must_equal true
+      cache_enabled_backend.backend.cache_enabled?(:command).must_equal true
+
+      cache_disabled_config = Inspec::Config.new(backend_cache: false)
+      cache_disabled_backend = Inspec::Backend.create(cache_disabled_config)
+      cache_disabled_backend.backend.cache_enabled?(:file).must_equal false
+      cache_disabled_backend.backend.cache_enabled?(:command).must_equal false
+    end
+
+    it 'enables caching when using a mock backend' do
+      backend.backend.cache_enabled?(:file).must_equal true
+      backend.backend.cache_enabled?(:command).must_equal true
```

Comment:
Can we change the name here so we don't have 'backend.backend' ?

Reply:
To what though? I also struggled because I'd have to use a custom one or change the `let(:backend)` block.